tests/validation/cattlevalidationtest/core/test_cluster.py

```diff
@@ -0,0 +1,136 @@
+            requestedHostId=hosts[1].id)
+        wait_for_condition(
+            client, socat_container2,
+            lambda x: x.state == 'running',
+            lambda x: 'State is: ' + x.state)
+
+        cluster = client.create_cluster(name=test_name + '-cluster',
+                                        port=9000)
+        cluster = wait_for_condition(
+            client, cluster,
+            lambda x: x.state == 'inactive',
+            lambda x: 'State is: ' + x.state)
+
+        cluster.addhost(hostId=hosts[0].id)
+        cluster = wait_for_condition(
+            client, cluster,
+            lambda x: len(x.hosts()) == 1,
+            lambda x: 'Number of hosts in cluster is: ' + len(x.hosts()))
+
+        cluster.activate()
+        cluster = wait_for_condition(
+            client, cluster,
+            lambda x: x.state == 'active',
+            lambda x: 'State is: ' + x.state)
+
+        # check Cluster Server is deployed
+        cluster_server = None
+        for instance in cluster.instances():
+            if (instance.state == 'removed'):
+                continue
+            if (instance.name == 'Cluster Server'):
+                cluster_server = instance
+                break
+
+        assert cluster_server is not None
+
+        assert cluster_server.systemContainer == 'ClusterAgent'
+        assert cluster_server.state == 'running'
+
+        cluster.addhost(hostId=hosts[1].id)
+        cluster = wait_for_condition(
+            client, cluster,
+            lambda x: len(x.hosts()) == 2,
+            lambda x: 'Number of hosts in cluster is: ' + len(x.hosts()))
+
+        # deploy container to cluster
+        test_container = client.create_container(
+            name=test_name + "-cl-deployed",
+            networkIds=[managed_network.id],
+            imageUuid='docker:ubuntu',
+            requestedHostId=cluster.id)
+        wait_for_condition(
+            client, test_container,
+            lambda x: x.state == 'running',
+            lambda x: 'State is: ' + x.state)
+
+        test_container_found = None
+        for instance in cluster.instances():
+            if (instance.name == test_name + '-cl-deployed'):
+                test_container_found = instance
+                break
+
+        assert test_container_found is not None
+        assert test_container.id == test_container_found.id
+
+        # check associated host is not a cluster
+        host_deployed_to = test_container_found.hosts()
+        assert host_deployed_to[0].kind == 'docker'
+
+        nic = super_client.list_container(uuid=test_container.uuid)[0].nics()[0]
+        assert nic.vnetId is not None
+        assert nic.subnetId is not None
+
+        # check primaryIpAddress is using the managed network (10.x.x.x) instead
+        # of using the docker IP (172.x.x.x)
+        # not a great check but better than nothing
+        primary_ip_address = test_container_found.data.fields.primaryIpAddress
```

Comment:
As a future enhancement, you could check the IP like we do here:

https://github.com/rancherio/validation-tests/blob/master/tests/validation/cattlevalidationtest/core/test_container.py#L120-L156